server/src/main/java/org/opensearch/search/fetch/FetchSearchResult.java

```diff
@@ -64,9 +72,24 @@ public FetchSearchResult(StreamInput in) throws IOException {
         hits = new SearchHits(in);
     }
 
+    public FetchSearchResult(InputStream in) throws IOException {
```

Comment:
This is a code smell. We have `org.opensearch.core.common.io.stream.StreamInput` and `java.io.InputStream`, and two constructors that do completely different things depending on whether we pass `StreamInput` or `InputStream`. We're lucky they are called different class names :) 

This needs to be disambiguated, likely by implementing a base class and separate implementations for one vs. the other stream.

Reply:
Both the constructors come from its base class implementing Writeable and BytesWriteable (this is not part of this PR). Let me see if I can make this a little more disambiguated.

Reply:
We need to do the same things as with Document Field - the class should not deal with all possible (de)serealizations